pymeasure/display/windows/dock_window.py

```diff
@@ -0,0 +1,200 @@
+            inputs_scroll.setWidget(self.inputs)
+            inputs_vbox.addWidget(inputs_scroll, 1)
+
+        else:
+            inputs_vbox.addWidget(self.inputs)
+
+        if self.directory_input:
+            inputs_vbox.addLayout(vbox)
+        else:
+            inputs_vbox.addLayout(hbox)
+
+        inputs_vbox.addStretch(0)
+        inputs_dock.setLayout(inputs_vbox)
+
+        dock = QtWidgets.QDockWidget('Input Parameters')
+        dock.setWidget(inputs_dock)
+        dock.setFeatures(QtWidgets.QDockWidget.DockWidgetFeature.NoDockWidgetFeatures)
+        self.addDockWidget(QtCore.Qt.DockWidgetArea.LeftDockWidgetArea, dock)
+
+        if self.use_sequencer:
+            sequencer_dock = QtWidgets.QDockWidget('Sequencer')
+            sequencer_dock.setWidget(self.sequencer)
+            sequencer_dock.setFeatures(QtWidgets.QDockWidget.NoDockWidgetFeatures)
+            self.addDockWidget(QtCore.Qt.DockWidgetArea.LeftDockWidgetArea, sequencer_dock)
+
+        if self.use_estimator:
+            estimator_dock = QtWidgets.QDockWidget('Estimator')
+            estimator_dock.setWidget(self.estimator)
+            estimator_dock.setFeatures(QtWidgets.QDockWidget.DockWidgetArea.NoDockWidgetFeatures)
+            self.addDockWidget(QtCore.Qt.DockWidgetArea.LeftDockWidgetArea, estimator_dock)
+
+        self.tabs = QtWidgets.QTabWidget(self.main)
+
+        self.dock_area = DockArea()
+        self.dock_area.name = 'Dock Tab'
+        self.docks = []
+
+        self.tabs.addTab(self.dock_area, self.dock_area.name)
+
+        for idx, i in enumerate(range(self.num_plots)):
+            x_axis_label = self.x_axis
+            y_axis_label = self.y_axis
+            # If x_axis or y_axis are a list, then we want to set the label to the passed list.
+            # However, if list is smaller than num_plots, repeat last item in the list.
+            if type(self.x_axis) == list:
+                x_axis_label = self.x_axis[min(idx, len(self.x_axis) - 1)]
+            if type(self.y_axis) == list:
+                y_axis_label = self.y_axis[min(idx, len(self.y_axis) - 1)]
+            self.widget_list.append(
+                PlotWidget("Results Graph", self.procedure_class.DATA_COLUMNS, x_axis_label,
+                           y_axis_label))
+            dock = Dock("Dock " + str(i + 1), closable=False, size=(200, 50))
+            self.dock_area.addDock(dock)
+            dock.addWidget(self.widget_list[i])
+            dock.nStyle = """
+                          Dock > QWidget {
+                              border: 1px solid #ff6600;
+                              border-radius: 5px;
+                          }"""
+            dock.dragStyle = """
+                          Dock > QWidget {
+                              border: 14px solid #ff6600;
+                              border-radius: 15px;
+                          }"""
+            dock.updateStyle()
+            self.docks.append(dock)
```

Comment:
So basically, as I suggest in my general comment, I think that this could be the basis of a separate `TabWidget` which can then be added in a ManagedWindowBase subclass (or maybe even it is not required to make a seperate class of this, but rather allow people to add it to the subclass of the ManagedWindow that you make for a specific experiment.)

Reply:
In last refactor I moved this logic to DockWidget